### PR TITLE
[MOD-13597] Add FFI functions for RLookupRow_Get/SetSortingVector

### PR DESCRIPTION
**Pull request by memark**
Add small FFI functions for `RLookupRow_GetSortingVector` and `RLookupRow_SetSortingVector`.

---

> [!NOTE]
> Introduces FFI accessors for a row’s sorting vector and aligns the Rust API to allow clearing it.
> 
> - Add `RLookupRow_GetSortingVector` and `RLookupRow_SetSortingVector` to `rlookup_ffi/src/row.rs` and expose them in `headers/rlookup_rs.h`
> - Change `rlookup::RLookupRow::set_sorting_vector` signature to `Option<&RSSortingVector<_>>`; update tests to call `set_sorting_vector(Some(&sv))`
> - Wire up `sorting_vector` dependency in `c_entrypoint/rlookup_ffi/Cargo.toml` and adjust imports (`ptr::{self, NonNull}`)
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit cf4ca07d12c920c57c73b17011be7f2425c8ccff. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs [hunk 1]
```diff
@@ -321,3 +321,44 @@ unsafe extern "C" fn RLookupRow_Get(
 
     row.get(key).map(|x| NonNull::new(x.as_ptr()).unwrap())
 }
+
+/// Returns the sorting vector (if any) for the row.
```
Comment (JonasKruckenberg) [inline src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs]:
```suggestion
/// Returns the sorting vector for the row, or `NULL` if none exists.
```

I like being explicit about the other return value, maybe we can phrase it like so?

File: src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs [hunk 2]
```diff
@@ -321,3 +321,44 @@ unsafe extern "C" fn RLookupRow_Get(
...
+) -> *const sorting_vector::RSSortingVector<RSValueFFI> {
+    // Safety: ensured by caller (1.)
+    let row = unsafe { row.as_ref().unwrap() };
+
+    row.sorting_vector()
+        .map(ptr::from_ref)
+        .unwrap_or(std::ptr::null())
+}
+
+/// Sets the sorting vector (if any) for the row.
```
Comment (JonasKruckenberg) [inline src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs]:
```suggestion
/// Sets the sorting vector for the row.
```

**Comment by memark** [comment]
Got a response from Redis that null is possible. Added that possibility back in the code.

File: src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs [hunk 3]
```diff
@@ -328,3 +328,44 @@ unsafe extern "C" fn RLookupRow_Get(
...
+/// Sets the sorting vector for the row.
+///
+/// # Safety
+///
+/// 1. `row` must be a [valid], non-null pointer to an [`RLookupRow`].
+/// 2. `sv` must be either null or a [valid], non-null pointer to an [`sorting_vector::RSSortingVector`].
+///
+/// [valid]: https://doc.rust-lang.org/std/ptr/index.html#safety
+#[unsafe(no_mangle)]
+const unsafe extern "C" fn RLookupRow_SetSortingVector(
```
[the remark this answers is not in the document]
Comment (oshadmi) [reply src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs:360]: @memark Can `const` be removed?

File: src/redisearch_rs/headers/rlookup_rs.h [hunk 4]
```diff
@@ -597,6 +597,30 @@ void RLookupRow_WriteFieldsFrom(const RLookupRow *src_row,
...
+/**
+ * Returns the sorting vector for the row, or null if none exists.
+ *
+ * # Safety
+ *
+ * 1. `row` must be a [valid], non-null pointer to an [`RLookupRow`].
+ *
+ * [valid]: https://doc.rust-lang.org/std/ptr/index.html#safety
+ */
+const RSSortingVector<RSValueFFI> *RLookupRow_GetSortingVector(const RLookupRow *row);
```
Comment (oshadmi) [inline src/redisearch_rs/headers/rlookup_rs.h:609]: Nitpick: Consider adding FFI test for `RLookupRow_GetSortingVector` and `RLookupRow_SetSortingVector` with null values (although the inner functionality is covered)

File: src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (memark) [reply src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs:360]: It could, but then we would have to disable the corresponding lint instead. It's good practice to mark functions as `const` in rust when they are, well, const. :) It doesn't carry over the FFI boundary, and will have no effect on the C side, so no harm there.

[the remark this answers is not in the document]
Comment (oshadmi) [reply src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs:360]: OK, only Rust related - they could be used in const compile time expressions,
So better to keep it? 👍🏼

File: src/redisearch_rs/headers/rlookup_rs.h (hunk 4 above)
Comment (memark) [reply src/redisearch_rs/headers/rlookup_rs.h:609]: Good question.

What the safety comment actually says/means is that passing null is Undefined Behaviour (UB). We do not guarantee what will happen if the caller does. Therefore I don't think it's meaningful to add a test for it. We would only test the current implementation (`unwrap` -> panic), but that is not necessarily how we want to handle it long term. It might be that `unwrap` is too slow and we want to change to `unwrap_unchecked`, which would then invalidate these tests.

File: src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs (hunk 3 above)
[the remark this answers is not in the document]
Comment (memark) [reply src/redisearch_rs/c_entrypoint/rlookup_ffi/src/row.rs:360]: Yes, exactly!
`const` in most other places concern pointers, and those are relevant on the C side. `const fn` is only for Rust side, so I also think it's good to keep it.
